### PR TITLE
Graphql review changes

### PR DESCRIPTION
**Pull request by piyushtiw**

### CODE REVIEW

**Comment by piyushtiw** [comment]
@rkr090 please review it.

File: app/controllers/graphql_controller.rb [hunk 1]
```diff
@@ -7,7 +7,6 @@ def execute
     result = AcehelpSchema.execute(query, variables: variables, context: context, operation_name: operation_name)
     render json: result
   rescue => e
-    raise e unless Rails.env.development?
     handle_error_in_development e
```
Comment (rkr090) [inline app/controllers/graphql_controller.rb]: This is no longer being only handled in development.

The method ```handle_error_in_development``` returns the backtrace in the JSON response. I don't think it is needed.

File: app/controllers/api/v1/base_controller.rb [hunk 2]
```diff
@@ -5,8 +5,6 @@ module V1
     class BaseController < ActionController::Base
       include ::Concerns::ErrorHandlers
 
-      skip_before_action :verify_authenticity_token
```
Comment (rkr090) [inline app/controllers/api/v1/base_controller.rb]: I have added my views here.
https://github.com/bigbinary/acehelp/pull/76#discussion_r195155603

**Comment by rkr090** [comment]
@vipulnsward _a please review

File: app/controllers/graphql_controller.rb [hunk 3]
```diff
@@ -48,10 +47,7 @@ def ensure_hash(ambiguous_param)
       end
     end
 
-    def handle_error_in_development(e)
-      logger.error e.message
-      logger.error e.backtrace.join("\n")
-
-      render json: { error: { message: e.message, backtrace: e.backtrace }, data: {} }, status: 500
+    def return_error(e)
```
Comment (vipulnsward) [inline app/controllers/graphql_controller.rb]: Why a new method? Is it being used somewhere else?

File: app/graphql/mutations/article_mutations.rb [hunk 4]
```diff
@@ -11,13 +11,12 @@ class Mutations::ArticleMutations
...
-    resolve ->(object, inputs, ctx) {
+    resolve ->(object, inputs, context) {
       category = Category.find_by_id(inputs[:category_id])
       return { errors: "Category not found" } if category.nil?
 
-      articles = category.articles
-      new_article = articles.build(title: inputs[:title], desc: inputs[:desc])
-      new_article.organization = ctx[:organization]
+      new_article = category.articles.build(title: inputs[:title], desc: inputs[:desc])
+      new_article.organization = context[:organization]
```
Comment (vipulnsward) [inline app/graphql/mutations/article_mutations.rb]: Can't pass organization to `build`?

File: app/graphql/mutations/article_mutations.rb [hunk 5]
```diff
@@ -62,11 +61,11 @@ class Mutations::ArticleMutations
...
     return_field :errors, types.String
 
-    resolve ->(_obj, inputs, ctx) {
-      article = Article.find_by(id: inputs[:id], organization_id: ctx[:organization].id)
+    resolve ->(_obj, inputs, context) {
+      article = Article.find_by(id: inputs[:id], organization_id: context[:organization].id)
       return { errors: "Article not found" } if article.nil?
 
-      article.destroy
+      article.destroy!
```
Comment (vipulnsward) [inline app/graphql/mutations/article_mutations.rb]: This is not what I meant in previous comment, I meant there should be a handler to return proper json response if destroy fails

if object.destroy
# do something
else 
# return error
end

File: app/graphql/mutations/url_mutations.rb [hunk 6]
```diff
@@ -52,11 +52,11 @@ class Mutations::UrlMutations
...
     return_field :errors, types.String
 
-    resolve ->(_obj, inputs, ctx) {
-      url = Url.find_by(id: inputs[:id], organization_id: ctx[:organization].id)
+    resolve ->(_obj, inputs, context) {
+      url = Url.find_by(id: inputs[:id], organization_id: context[:organization].id)
       return { errors: "Url not found" } if url.nil?
 
-      url.destroy
+      url.destroy!
```
Comment (vipulnsward) [inline app/graphql/mutations/url_mutations.rb]: ditto

**Comment by piyushtiw** [comment]
@vipulnsward _a please review it.

File: app/controllers/graphql_controller.rb [hunk 7]
```diff
@@ -7,8 +7,7 @@ def execute
     result = AcehelpSchema.execute(query, variables: variables, context: context, operation_name: operation_name)
     render json: result
   rescue => e
-    raise e unless Rails.env.development?
-    handle_error_in_development e
+    render json: { error:  e.message }, status: 500
```
Comment (vipulnsward) [inline app/controllers/graphql_controller.rb]: @rkr090 can you investigate more what are error handling conventions from Graphql API's. Simply spitting out errors may not be desirable

**Pull request merged**